src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IParenthesizedExpression.vb

```diff
@@ -0,0 +1,391 @@
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of ParenthesizedExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact>
+        Public Sub TestParenthesizedConversionParent()
+            Dim source = <![CDATA[
+Class P
+    Private Shared Function M1(a As Integer, b As Integer) As Long
+        Return (a + b)'BIND:"Return (a + b)"
+    End Function
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IReturnStatement (OperationKind.ReturnStatement) (Syntax: 'Return (a + b)')
+  ReturnedValue: 
+    IConversionExpression (Implicit, TryCast: False, Unchecked) (OperationKind.ConversionExpression, Type: System.Int64, IsImplicit) (Syntax: '(a + b)')
+      Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: True, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+      Operand: 
+        IParenthesizedExpression (OperationKind.ParenthesizedExpression, Type: System.Int32) (Syntax: '(a + b)')
+          Operand: 
+            IBinaryOperatorExpression (BinaryOperatorKind.Add, Checked) (OperationKind.BinaryOperatorExpression, Type: System.Int32) (Syntax: 'a + b')
+              Left: 
+                IParameterReferenceExpression: a (OperationKind.ParameterReferenceExpression, Type: System.Int32) (Syntax: 'a')
+              Right: 
+                IParameterReferenceExpression: b (OperationKind.ParameterReferenceExpression, Type: System.Int32) (Syntax: 'b')
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of ReturnStatementSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact>
+        Public Sub TestParenthesizedConstantValue()
+            Dim source = <![CDATA[
+Class P
+    Private Shared Function M1() As Integer
+        Return (5)'BIND:"(5)"
+    End Function
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IParenthesizedExpression (OperationKind.ParenthesizedExpression, Type: System.Int32, Constant: 5) (Syntax: '(5)')
+  Operand: 
+    ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 5) (Syntax: '5')
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of ParenthesizedExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact(Skip:="https://github.com/dotnet/roslyn/issues/22879")>
+        Public Sub TestParenthesizedDelegateCreation()
+            Dim source = <![CDATA[
+Class P
+    Private Shared Function M1() As Object
```

Comment:
>Object [](start = 36, length = 6)

It would be good to also test scenario when the type is a delegate type.

Reply:
and the conversion is implicit.

---
In reply to: [147803696](https://github.com/dotnet/roslyn/pull/22846#discussion_r147803696) [](ancestors = 147803696)